library/std/src/env.rs

```diff
@@ -582,7 +589,7 @@ impl Error for JoinPathsError {
 /// ```
 #[deprecated(
     since = "1.29.0",
-    note = "This function's behavior is unexpected and probably not what you want. \
+    note = "This function's behavior may be unexpected on Windows. \
```

Comment:
Maybe it would be better to not mention Windows in the deprecation note. Otherwise users might think, that it is fine on the current non-windows platform they develop for, so they suppress the lint. This will lead to platform-dependent code while not strictly necessary.

Reply:
But it *is* fine on non-windows platforms. I'm not keen on deliberately misleading users.